warplib/src/warp_memory/arrays.cairo

```diff
@@ -0,0 +1,93 @@
+use warplib::warp_memory::WarpMemory;
+use warplib::warp_memory::WarpMemoryTrait;
+use warplib::warp_memory::WarpMemoryImpl;
+use integer::u256_from_felt252;
+use integer::u128_try_from_felt252;
+use integer::Into;
+use option::OptionTrait;
+
+trait WarpMemoryArraysTrait {
+    /// reads the pointer of an array. If it does not exists, it creates a new one
+    fn read_id(ref self: WarpMemory, location: felt252, size: felt252) -> felt252;
```

Comment:
It's rather `read_or_create` . `read_id` could be a separate method

Reply:
it creates an empty pointer for reading solidity uninitialized memory. It does not create anything meaningful, also `read_id` is all over the code. I think `read_or_create` can be misleading. `read_id` gets a pointer to an array, if there is none, then it returns a default one.

Reply:
Imo `read_id` is misleading since nobody expects `read_id` to write to the memory. Maybe `read_or_initialize`?

Reply:
Yeah you are right actually, let's use `get_or_create_id`